src/qibojit/backends/cpu.py

```diff
@@ -236,12 +236,19 @@ def _apply_ygate_density_matrix(self, gate, state, nqubits):
 
     def apply_channel_density_matrix(self, channel, state, nqubits):
         state = self.cast(state)
+        if channel.name == "ReadoutErrorChannel":
+            state_copy = self.cast(state, copy=True)
         new_state = (1 - channel.coefficient_sum) * state
         for coeff, gate in zip(channel.coefficients, channel.gates):
             state = self.apply_gate_density_matrix(gate, state, nqubits)
             new_state += coeff * state
             # reset the state
-            state = self.apply_gate_density_matrix(gate, state, nqubits, inverse=True)
+            if channel.name == "ReadoutErrorChannel":
+                state = state_copy
```

Comment:
Have you tested this with an example that there are more than two terms in the Kraus sum? I suspect that here you will lose the reference to one of the state copies and it will not work for the third term and on. I think the copy needs to be done inside the loop, but I am not 100% sure.

Reply:
I see what you mean. Changed this by adding an extra `self.cast(state_copy, copy=True`